lib/handler/connect.c

```diff
@@ -708,6 +714,9 @@ static h2o_iovec_t udp_get_next_chunk(const char *start, size_t len, size_t *to_
      * using a CONNECT-UDP Stream Chunk Type of UDP_PACKET (value 0x00).
      */
     *skip = chunk_type != 0;
+    if (datagram_format == H2O_DATAGRAM_FORMAT_RFC) {
+        *skip |= capsule_context_id != 0;
+    }
     *to_consume = (bytes + chunk_length) - (const uint8_t *)start;
 
     return h2o_iovec_init(bytes, chunk_length);
```

Comment:
Same here, the `chunk_length` includes the context id field yet `bytes` have moved past it.  Please see doohara/h2o#1.  (Details: we need to consume the full capsule from the stream even though `bytes` has advanced into the capsule, and the length of the returned payload needs to not include the length of the context id because `bytes` have already advanced past that field.)